src/test/java/tatracker/model/group/GroupTest.java

```diff
@@ -0,0 +1,178 @@
+        Group group = MANY_STUDENTS;
+        group.sortStudentsByRatingDescending();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.ELLE);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.BENSON);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.FIONA);
+    }
+
+    @Test
+    public void sortStudentsByMatricNumber() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsByMatricNumber();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.BENSON);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.ELLE);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.FIONA);
+    }
+
+    @Test
+    public void hasStudent() {
+        //First student in student list
+        assertTrue(MANY_STUDENTS.hasStudent(TypicalStudents.ALICE));
+
+        //Middle of the list
+        assertTrue(MANY_STUDENTS.hasStudent(TypicalStudents.ELLE));
+
+        //End of the list
+        assertTrue(MANY_STUDENTS.hasStudent(TypicalStudents.DANIEL));
+
+        //Not in the list
+        assertFalse(MANY_STUDENTS.hasStudent(TypicalStudents.BOB));
+
+        //List has no students
+        assertFalse(NO_STUDENTS.hasStudent(TypicalStudents.ALICE));
+
+        //Only one in the list
+        assertTrue(ONE_STUDENT.hasStudent(TypicalStudents.ALICE));
+
+        //Not in the list
+        assertFalse(ONE_STUDENT.hasStudent(TypicalStudents.BOB));
+    }
+
+    @Test
+    public void getIdentifier() {
+        assertEquals("G03", ONE_STUDENT.getIdentifier());
+    }
+
+    @Test
+    public void getGroupType() {
+        assertEquals(GroupType.LAB, ONE_STUDENT.getGroupType());
+        assertEquals(GroupType.TUTORIAL, NO_STUDENTS.getGroupType());
+        assertEquals(GroupType.OTHER, MANY_STUDENTS.getGroupType());
+    }
+
+    @Test
+    public void getStudent() {
+        assertEquals(TypicalStudents.DANIEL, MANY_STUDENTS.getStudent(new Matric("A0192154M")));
+    }
+
+    @Test
+    public void addStudent() {
+        Group group = new Group("G03");
+        group.addStudent(TypicalStudents.BOB);
+        assertTrue(group.hasStudent(TypicalStudents.BOB));
+        group.addStudent(TypicalStudents.ALICE);
+        assertTrue(group.hasStudent(TypicalStudents.ALICE));
+        group.addStudent(TypicalStudents.DANIEL);
+        assertTrue(group.hasStudent(TypicalStudents.BOB));
+        assertTrue(group.hasStudent(TypicalStudents.ALICE));
+        assertTrue(group.hasStudent(TypicalStudents.DANIEL));
+    }
+
+    @Test
+    public void deleteStudent() {
+        Group group = new Group("G03");
+        group.addStudent(TypicalStudents.BOB);
+        group.addStudent(TypicalStudents.ALICE);
+        group.addStudent(TypicalStudents.DANIEL);
+
+        group.deleteStudent(TypicalStudents.BOB);
+        assertFalse(group.hasStudent(TypicalStudents.BOB));
```

Comment:
assert Alice and Daniel inside

Reply:
Unnecessary. Shouldn't add too many tests - expensive